decode-json.go

```diff
@@ -0,0 +1,25 @@
+package tfjs
+
+import (
+	"io/ioutil"
+	"os"
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+// ExtractPlanStruct extracts tfjson.Plan struct from file in provided path
+// and returns the pointer on it if it is possible, otherwise return error.
+func ExtractPlanStruct (filePath string) (*tfjson.Plan, error) {
+	f, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	
+	byteFile, _ := ioutil.ReadAll(f)
+
+	var plan tfjson.Plan
+	if err = plan.UnmarshalJSON(byteFile); err != nil {
+		return nil, err
+	}
+	return &plan, nil
```

Comment:
I think that returning the address of a variable declared on stack in the function is not a good idea. When calling other functions or doing stuff with the stack, the variable "plan" here might get overwritten. I think it's best to allocate memory dynamically for it.